internal/pkg/skuba/kubernetes/node_version_test.go

```diff
@@ -18,394 +18,1027 @@
+		clusterVersion   version.Version
+		expectReturn     bool
+	}{
+		{
+			name:             "master node version less than cluster version",
+			isMaster:         true,
+			apiServerVersion: clusterVersionMinus,
+			kubeletVersion:   clusterVersionMinus,
+			clusterVersion:   clusterVersion,
+			expectReturn:     true,
+		},
+		{
+			name:             "master node (kubelet) version less than cluster version",
+			isMaster:         true,
+			apiServerVersion: clusterVersion,
+			kubeletVersion:   clusterVersionMinus,
+			clusterVersion:   clusterVersion,
+			expectReturn:     true,
+		},
+		{
+			name:             "master node version more than cluster version",
+			isMaster:         true,
+			apiServerVersion: clusterVersionPlus,
+			kubeletVersion:   clusterVersionPlus,
+			clusterVersion:   clusterVersion,
+			expectReturn:     false,
+		},
+		{
+			name:           "worker node version less than cluster version",
+			kubeletVersion: clusterVersionMinus,
+			clusterVersion: clusterVersion,
+			expectReturn:   true,
+		},
+		{
+			name:           "worker node version more than cluster version",
+			kubeletVersion: clusterVersionPlus,
+			clusterVersion: clusterVersion,
+			expectReturn:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var nodeVersionInfo NodeVersionInfo
+			switch tt.isMaster {
+			case true:
+				nodeVersionInfo = NodeVersionInfo{
+					APIServerVersion: &tt.apiServerVersion,
+					KubeletVersion:   &tt.kubeletVersion,
+				}
+			default:
+				nodeVersionInfo = NodeVersionInfo{
+					KubeletVersion: &tt.kubeletVersion,
+				}
+			}
+			actualReturn := nodeVersionInfo.LessThanClusterVersion(&tt.clusterVersion)
+			if actualReturn != tt.expectReturn {
+				t.Errorf("got: (%v), want: (%v)", actualReturn, tt.expectReturn)
+			}
+		})
+	}
+}
+
+func TestDriftsFromClusterVersion(t *testing.T) {
+	clusterVersion := *version.MustParseSemantic("v1.1.1")
+	majorLess := *version.MustParseSemantic("v0.1.1")
+	minorLess := *version.MustParseSemantic("v1.0.1")
+	tests := []struct {
+		name             string
+		isMaster         bool
+		apiServerVersion version.Version
+		kubeletVersion   version.Version
+		clusterVersion   version.Version
+		expectReturn     bool
+	}{
+		{
+			name:             "master node version same as cluster version",
+			isMaster:         true,
+			apiServerVersion: clusterVersion,
+			kubeletVersion:   clusterVersion,
+			clusterVersion:   clusterVersion,
+			expectReturn:     false,
+		},
+		{
+			name:             "master node (apiserver) major version less than cluster version",
+			isMaster:         true,
+			apiServerVersion: majorLess,
+			clusterVersion:   clusterVersion,
+			expectReturn:     true,
+		},
+		{
+			name:             "master node (apiserver) minor version less than cluster version",
+			isMaster:         true,
+			apiServerVersion: minorLess,
+			clusterVersion:   clusterVersion,
+			expectReturn:     true,
+		},
```

Comment:
`kubeletVersion` is necessary for master node, right? [1]

Reply:
This case expects to return before check for kubeletVersion.
```
if nvi.IsControlPlane() {
		if nvi.APIServerVersion.Major() < clusterVersion.Major() ||
			nvi.APIServerVersion.Minor() < clusterVersion.Minor() {
			return true
		}
	}
```

Reply:
Yes, I knew it.

I thought the master node components _should_ includes apiserver and kubelet. (I knew the kubelet won't be used for this test case, the idea coming from the master node component point of view, make sense?)

Reply:
Unit-test is meant to test code and only adding what is necessary, it does not meant to go beyond where it meant to return.